packages/react-core/src/demos/examples/AlertGroup/AlertGroupToastWithNotificationDrawer.tsx

```diff
@@ -0,0 +1,365 @@
+    );
+  };
+
+  const markAllNotificationsRead = () => {
+    setNotifications(prevNotifications =>
+      prevNotifications.map(notification => ({ ...notification, isNotificationRead: true }))
+    );
+  };
+
+  const getUnreadNotificationsNumber = () =>
+    notifications.filter(notification => notification.isNotificationRead === false).length;
+
+  const onNotificationBadgeClick = () => {
+    removeAllAlerts();
+    setDrawerExpanded(!isDrawerExpanded);
+  };
+
+  const onDropdownToggle = (id: React.Key, isActive: boolean) => {
+    if (isActive) {
+      setOpenDropdownKey(id);
+      return;
+    }
+    setOpenDropdownKey(null);
+  };
+
+  const onDropdownSelect = () => {
+    setOpenDropdownKey(null);
+  };
+
+  const buildOverflowMessage = () => {
+    const overflow = alerts.length - maxDisplayed;
+    if (overflow > 0 && maxDisplayed > 0) {
+      return `View ${overflow} more notification(s) in notification drawer`;
+    }
+    return '';
+  };
+
+  const getUniqueId = () => new Date().getTime();
+
+  const getTimeCreated = () => {
+    const dateCreated = new Date();
+    return (
+      dateCreated.toDateString() +
+      ' at ' +
+      ('00' + dateCreated.getHours().toString()).slice(-2) +
+      ':' +
+      ('00' + dateCreated.getMinutes().toString()).slice(-2)
+    );
+  };
+
+  const removeAlert = (key: React.Key) => {
+    setAlerts(prevAlerts => prevAlerts.filter(alert => alert.props.id !== key.toString()));
+  };
+
+  const removeAllAlerts = () => {
+    setAlerts([]);
+  };
+
+  const onAlertGroupOverflowClick = () => {
+    removeAllAlerts();
+    setDrawerExpanded(true);
+  };
+
+  const onMaxDisplayedAlertsMinus = () => {
+    setMaxDisplayed(normalizeAlertsNumber(maxDisplayed - 1));
+  };
+
+  const onMaxDisplayedAlertsChange = (event: any) => {
+    setMaxDisplayed(normalizeAlertsNumber(Number(event.target.value)));
+  };
+
+  const onMaxDisplayedAlertsPlus = () => {
+    setMaxDisplayed(normalizeAlertsNumber(maxDisplayed + 1));
+  };
+
+  const normalizeAlertsNumber = (value: number) => Math.max(Math.min(value, maxAlerts), minAlerts);
+
+  const alertButtonClasses = 'pf-c-button pf-m-secondary';
+
+  const alertButtonStyle = { marginRight: '8px', marginTop: '8px' };
+
+  const notificationBadge = (
+    <NotificationBadge
+      variant={getUnreadNotificationsNumber() === 0 ? 'read' : 'unread'}
+      onClick={onNotificationBadgeClick}
+      aria-label="Notifications"
+    >
+      <BellIcon />
+    </NotificationBadge>
+  );
+
+  const notificationDrawerActions = [
+    <DropdownItem key="markAllRead" onClick={markAllNotificationsRead} component="button">
+      Mark all read
+    </DropdownItem>,
+    <DropdownItem key="clearAll" onClick={removeAllNotifications} component="button">
+      Clear all
+    </DropdownItem>
+  ];
+
+  const notificationDrawerDropdownItems = (key: React.Key) => [
+    <DropdownItem key="markRead" component="button" onClick={() => markNotificationRead(key)}>
+      Mark as read
```

Comment:
FWIW the existing demos are marked as read on click/enter - should this match?

Reply:
From the design doc:
"The user can individually select “Mark as read” or “Clear” on the new notifications from the alert group if they like."
[https://docs.google.com/document/d/1sW3b6wJNDCeZaxNQv895APoqZROz6BPRyBNj2D4Rmco/edit#heading=h.802igegr9dp3](https://docs.google.com/document/d/1sW3b6wJNDCeZaxNQv895APoqZROz6BPRyBNj2D4Rmco/edit#heading=h.802igegr9dp3)
It didn't make sense to me to use both on click/enter and Kebab, so I used only the second one

Reply:
The design guidelines here do specify that the user can make an alert as read either by clicking the list entry or from the kabob menu: https://www.patternfly.org/v4/components/notification-drawer/design-guidelines While it is redundant, I think the idea was to provide a simple one-click way to mark items as read. @tompsota if you can update this, then great. But I also don't feel like it's critical since this demo is not really about demonstrating the notification drawer.

Reply:
Updated